### PR TITLE
fix: use `RecordBuilder` variants of result construction in transfer HtsCalls

### LINKED ISSUE #10424: `LazyCreateThroughPrecompileSuite.erc20TransferFromLazyCreate()`

### LINKED ISSUE #10431: `LazyCreateThroughPrecompileSuite.resourceLimitExceededRevertsAllRecords()`

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #10424 by returning the `RecordBuilder` from an ERC transfer dispatch to the `MessageCallProcessor` so it can be updated with a status of `INSUFFICIENT_GAS`.
 - Closes #10431 by adding `ifHapiTest()` and `ifNotHapiTest()` spec operations to target the different behavior mod-service will exhibit in the edge case that insufficient child records are available to handle a contract operation.
    * Instead of throwing an exception to the top-level HAPI transaction, mod-service preserves a "clean shutdown" of the EVM transaction w/ traceability information.
 - _Misc_: 
    * Extracts an `inProgressBody()` method in `SingleTransactionRecordBuilderImpl` to improve readability and possible reuse.
    * Deletes the unused `execute()` overload in `Erc721TransferFromCall`.

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleResult.java [hunk 1]
```diff
@@ -37,6 +37,25 @@
...
+ * <ul>
+ *     <li>{@link #payerKey}</li>
+ *     <li>{@link #status}</li>
+ *     <li>{@link #responseCode}</li>
+ *     <li>{@link #requiredKeys}</li>
+ *     <li>{@link #optionalKeys}</li>
+ *     <li>{@link #hollowAccounts}</li>
+ *     <li>{@link #verificationResults}</li>
+ *     <li>{@link #configVersion}</li>
+ * </ul>
```
Comment (lukelee-sl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleResult.java:57]: Is this related to the ISS you mentioned?  If so a note on why this information is important would be nice.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java [hunk 2]
```diff
@@ -239,6 +241,14 @@ public static <T> RecordSystemProperty<T> recordSystemProperty(
         return new RecordSystemProperty<>(property, converter, historian);
     }
 
+    public static NetworkTypeFilterOp ifHapiTest(@NonNull final HapiSpecOperation... ops) {
```
Comment (lukelee-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java:244]: Oh nice. This will likely come in very handy.

**Review by lukelee-sl — DISMISSED** [review]
LGTM.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleResult.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleResult.java:57]: Ah will elaborate in next PR.

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @tinker-michaelj

**Pull request merged**